libvisual/libvisual/lv_bmp.c

```diff
@@ -72,7 +73,7 @@ static int load_uncompressed (FILE *fp, VisVideo *video, int depth)
 			while (data > (uint8_t *) visual_video_get_pixels (video)) {
 				/* Unpack 4 bpp pixels aka 2 pixels per byte */
 				uint8_t *col = data - video->pitch;
-				uint8_t *end = (uint8_t *) ((int)data & ~1);
+				uint8_t *end = (uint8_t *) ((uintptr_t)data & ~1);
```

Comment:
Wow, this is atrocious. This arithmetic trick also depends on memory alignment of the `data` pointer to be 8 bytes or more, but this assumption seems to be relatively safe.